src/components/shared/Login.jsx

```diff
@@ -1,54 +1,93 @@
 
 const clientId = process.env.REACT_APP_CLIENTID;
 
 const Login = (props) => {
   const [showLoginButton, setShowLoginButton] = useState(true);
   const [showLogoutButton, setShowLogoutButton] = useState(false);
 
-  const onSuccess = (res) => {
-    console.log("Login Success! Current user: ", res);
-    window.localStorage.setItem("accessToken", res.accessToken);
-    props.setuserProfile(res.profileObj);
+  useEffect(() => {
+    if (props.isLogedin) {
+      setShowLoginButton(false);
+      setShowLogoutButton(true);
+    } else {
+      setShowLoginButton(true);
+      setShowLogoutButton(false);
+    }
+  }, [props.isLogedin]);
+
+  const onLoginSuccess = (res) => {
+    console.log(res, "RES");
+    window.localStorage.setItem("accessToken", res.token);
+    window.localStorage.setItem("userName", res.userData.name);
     props.setIsLogedin(true);
-    setShowLoginButton(false);
-    setShowLogoutButton(true);
+  };
+  // Google Auth START
+  const onGoogleAuthSuccess = (res) => {
+    let userData = { token: res.accessToken, userData: res.profileObj };
+    onLoginSuccess(userData);
   };
 
-  const onFailure = (res) => {
+  const onGoogleAuthFailure = (res) => {
     console.log("Login Failed! res:", res);
   };
+  // Google Auth ENDS
 
-  const onLogoutSuccess = (res) => {
-    console.log("Logout Success!");
+  const onLogoutSuccess = () => {
     props.setIsLogedin(false);
     window.localStorage.removeItem("accessToken");
-    setShowLoginButton(true);
-    setShowLogoutButton(false);
   };
 
+  const loginApiAuth = (data) => {
+    axios
+      .post("https://todo-node-api.vercel.app/api/login", {
+        username: data.email,
+        password: data.password,
+      })
+      .then(function (response) {
+        let userData = { token: response.data.token, userData: response.data };
+        onLoginSuccess(userData);
+      })
+      .catch(function (error) {
+        console.log(error);
+      });
+  };
+  console.log(showLoginButton, showLogoutButton);
   return (
-    <div class="button-login">
-      {showLoginButton ? (
-        <GoogleLogin
-          clientId={clientId}
-          buttonText="Login with your Gmail"
-          onSuccess={onSuccess}
-          onFailure={onFailure}
-          cookiePolicy={"single_host_origin"}
-          isSignedIn={true}
-        />
-      ) : null}
-
+    <>
       {showLogoutButton ? (
-        <GoogleLogout
-          clientId={clientId}
-          buttonText="Logout"
-          onLogoutSuccess={onLogoutSuccess}
-        />
+        <span class="button-google">
+          <GoogleLogout
+            clientId={clientId}
+            buttonText="Logout"
+            onLogoutSuccess={onLogoutSuccess}
+            className="button-logout"
+          />
+        </span>
       ) : null}
-    </div>
+
+      <div className="login-form">
+        {showLoginButton ? (
+          <div>
+            <LoginForm loginFunctionality={(data) => loginApiAuth(data)} />
```

Comment:
We can keep the whole login thing in the same component.

Reply:
Moved to same component